docs/api-reference/gltf-loaders/glb-builder.md

```diff
@@ -0,0 +1,79 @@
+# `GLBBuilder` class (@loaders.gl/gltf)
+
+The `GLBBuilder` class allows applications to use a "fluent" API to dynamically build up a hybrid JSON/binary GLB file. The `GLBBuilder` would normally be used if you want to save custom mixed JSON/binary data in a "GLB envelope".
+
+References:
+* For more information, see [glTF and GLB support]() in the Developer's Guide.
+
+
+## Usage
+
+Adding binary data sub chunks to the GLB file, then calling encode to generate the complete `arrayBuffer`.
+
+```js
+import {GLBBuilder} from '@loaders.gl/gltf';
+import {saveBinaryFile} from '@loaders.gl/core';
+
+const gltfBuilder = new GLBBuilder();
+
+const imageIndex = gltfBuilder.addImage();
+
+// Add custom JSON in top-level glTF object
+gltfBuilder.addApplicationData('app-key', {...});
+
+// All data added, we can encode
+const arrayBuffer = gltfBuilder.encodeAsGLB();
+
+// The encoded `ArrayBuffer` represents a complete image of the data
```

Comment:
use of the word "image" here is a bit confusing with an actual image.  Yes, pedantic nit.